src/round.rs

```diff
@@ -332,13 +335,23 @@ impl<Id, H, N, Signature> Round<Id, H, N, Signature> where
 						chain,
 					)?;
 
+					// Push the vote into HistoricalVotes.
+					let message = Message::Prevote(vote.clone());
+					let signed_message = SignedMessage { id: signer.clone(), signature, message };
+					self.historical_votes.push_vote(signed_message);
+
 					None
 				}
 				VoteMultiplicity::Equivocated(ref first, ref second) => {
 					// mark the equivocator as such. no need to "undo" the first vote.
 					self.bitfield_context.equivocated_prevote(info)
 						.expect("info is instantiated from same voter set as bitfield; qed");
 
+					// Push the vote into HistoricalVotes.
+					let message = Message::Prevote(vote);
+					let signed_message = SignedMessage { id: signer.clone(), signature, message };
+					self.historical_votes.push_vote(signed_message);
```

Comment:
Maybe only add if `vote == second`? Or maybe if `vote == first || vote == second` just to be safe wrt ordering.

Reply:
This is not needed since we early-exit after we've already seen an equivocation.